src/app/modules/startup/controller.nim

```diff
@@ -509,51 +488,40 @@ proc isSelectedAccountAKeycardAccount*(self: Controller): bool =
 
-proc loginLocalPairingAccount*(self: Controller) =
-  self.delegate.moveToLoadingAppState()
-  if self.localPairingStatus.chatKey.len == 0:
-    self.accountsService.login(self.localPairingStatus.account, self.localPairingStatus.password)
+  var passwordHash, chatPrivateKey, mnemonic = ""
+
+  if not keycard:
+    passwordHash = hashPassword(self.tmpPassword) 
   else:
-    var kcEvent = KeycardEvent()
-    kcEvent.keyUid = self.localPairingStatus.account.keyUid
-    kcEvent.whisperKey.privateKey = self.localPairingStatus.chatKey
-    kcEvent.encryptionKey.publicKey = self.localPairingStatus.password
-    discard self.accountsService.loginAccountKeycard(self.localPairingStatus.account, kcEvent)
-
-proc loginAccountKeycard*(self: Controller, storeToKeychainValue: string, keycardReplacement = false) =
-  if keycardReplacement:
+    passwordHash = self.tmpKeycardEvent.encryptionKey.publicKey
+    chatPrivateKey = self.tmpKeycardEvent.whisperKey.privateKey
+    mnemonic = self.tmpSeedPhrase
+
+  if keycard and keycardReplacement:
     self.delegate.applyKeycardReplacementAfterLogin()
-  singletonInstance.localAccountSettings.setStoreToKeychainValue(storeToKeychainValue)
-  self.delegate.moveToLoadingAppState()
-  let selAcc = self.getSelectedLoginAccount()
-  let error = self.accountsService.loginAccountKeycard(selAcc, self.tmpKeycardEvent)
-  if(error.len > 0):
-    self.delegate.emitAccountLoginError(error)
-
-proc loginAccountKeycardUsingSeedPhrase*(self: Controller, storeToKeychain: bool) =
-  let acc = self.accountsService.createAccountFromMnemonic(self.getSeedPhrase(), includeEncryption = true, includeWhisper = true)
-  let selAcc = self.getSelectedLoginAccount()
-
-  var kcData = KeycardEvent(
-    keyUid: acc.keyUid,
-    masterKey: KeyDetails(address: acc.address),
-    whisperKey: KeyDetails(privateKey: acc.derivedAccounts.whisper.privateKey),
-    encryptionKey: KeyDetails(publicKey: acc.derivedAccounts.encryption.publicKey)
+      
+  self.accountsService.login(
+    self.getSelectedLoginAccount(),
+    passwordHash,
+    chatPrivateKey,
+    mnemonic,
   )
-  if acc.derivedAccounts.whisper.privateKey.startsWith("0x"):
-    kcData.whisperKey.privateKey = acc.derivedAccounts.whisper.privateKey[2..^1]
-
-  self.setupKeychain(storeToKeychain)
 
+proc loginLocalPairingAccount*(self: Controller) =
   self.delegate.moveToLoadingAppState()
-  let error = self.accountsService.loginAccountKeycard(selAcc, kcData)
-  if(error.len > 0):
-    self.delegate.emitAccountLoginError(error)
+  self.accountsService.login(
+    self.localPairingStatus.account,
+    self.localPairingStatus.password,
+    chatPrivateKey = self.localPairingStatus.chatKey
+  )
+
+# FIXME: Why do we even have storeToKeychain during login? Makes no sense
+proc loginAccountKeycard*(self: Controller, storeToKeychain: bool, keycardReplacement = false) =
+  # singletonInstance.localAccountSettings.setStoreToKeychainValue(storeToKeychainValue)
```

Comment:
Is the comment and FIXME still needed?

Reply:
I left it to be done in https://github.com/status-im/status-desktop/issues/15167.
Will note this in the code comment.

Reply:
caf2b8daff0e98fe1c2c6804cf125e1ba3ec99a3